File: csharp/078_Subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        List<IList<int>> result = new List<IList<int>>();

        result.Add(new List<int>());
        foreach (int num in nums) {
            List<IList<int>> newResult = new List<IList<int>>();
            foreach (IList<int> prev in result) {
                List<int> newSet = new List<int>(prev);
                newSet.Add(num);
                newResult.Add(newSet);
            }
            result.AddRange(newResult);
        }

        return result;
    }
}

File: cpp/078_Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> result;
        
        result.push_back({});
        for (auto num : nums) {
            vector<vector<int>> newResult;
            for (auto prev : result) {
                vector<int> newSet(prev);
                newSet.push_back(num);
                newResult.push_back(newSet);
            }
            result.insert(result.end(), newResult.begin(), newResult.end());
        }

        return result;
    }
};